@here/harp-mapview/lib/TextureLoader.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as THREE from "three";
+
+interface RequestHeaders {
+    [field: string]: string;
+}
+
+/**
+ * A texture loader that also supports request headers(e.g. Authorization)
+ */
+export class TextureLoader extends THREE.TextureLoader {
+    /**
+     * Request headers
+     */
+    requestHeaders?: RequestHeaders;
+
+    /**
+     * Sets request headers
+     * @param headers
+     */
+    setRequestHeaders(requestHeaders: RequestHeaders | undefined): void {
+        this.requestHeaders = requestHeaders;
+    }
+
+    load(
+        url: string,
+        onLoad?: (texture: THREE.Texture) => void,
+        onProgress?: (event: ProgressEvent) => void,
+        onError?: (event: ErrorEvent) => void
+    ): THREE.Texture {
+        // Use normal texture loader if no request header is set
+        if (this.requestHeaders === undefined) {
+            return super.load(url, onLoad, onProgress, onError);
+        }
+
+        if (this.path !== undefined) {
+            url = this.path + url;
+        }
+        url = this.manager.resolveURL(url);
+
+        // Load image with fetch API if request header is set
+        const texture = new THREE.Texture();
+
+        this.manager.itemStart(url);
+        fetch(url, {
+            headers: this.requestHeaders,
+            mode: this.crossOrigin !== undefined ? "cors" : "no-cors"
+        })
+            .then(response => {
```

Comment:
is this formatted correctly? Seems weird.

Reply:
I also would like to format it differently but it's what prettier wants.
This would look much nicer:
```javascript
return textureLoader.load({
    //...
}).then(() => {
    //...
});
```